tests/unit/vault/test_strategy_management.py

```diff
@@ -130,8 +131,167 @@ def test_migrate_strategy__with_existing_debt(
+    locked_strategy.freeLockedFunds(sender=gov)
+
+    old_strategy_params = vault.strategies(old_strategy)
+    old_current_debt = old_strategy_params.current_debt
+    old_max_debt = old_strategy_params.max_debt
+
+    snapshot = chain.pending_timestamp
+    tx = vault.migrate_strategy(new_strategy.address, old_strategy.address, sender=gov)
+
+    event = list(tx.decode_logs(vault.StrategyMigrated))
+
+    assert len(event) == 1
+    assert event[0].old_strategy == old_strategy.address
+    assert event[0].new_strategy == new_strategy.address
+
+    new_strategy_params = vault.strategies(new_strategy)
+    assert new_strategy_params.activation == pytest.approx(snapshot, abs=1)
+    assert new_strategy_params.current_debt == old_current_debt
+    assert new_strategy_params.max_debt == old_max_debt
+    assert new_strategy_params.last_report == pytest.approx(snapshot, abs=1)
+
+    old_strategy_params = vault.strategies(old_strategy)
+    checks.check_revoked_strategy(old_strategy_params)
+
+
+def test_migrate_lossy_strategy__with_existing_debt__migrates(
+    gov,
+    asset,
+    user_deposit,
+    create_vault,
+    mint_and_deposit_into_vault,
+    create_lossy_strategy,
+    add_strategy_to_vault,
+    add_debt_to_strategy,
+    fish_amount,
+    fish
+):  
+    vault = create_vault(asset)
+    lossy_strategy = create_lossy_strategy(vault)
+    amount = fish_amount
+    shares = amount
+
+    vault.set_role(gov.address, ROLES.STRATEGY_MANAGER | ROLES.DEBT_MANAGER, sender=gov)
+    user_deposit(fish, vault, asset, amount)
+    add_strategy_to_vault(gov, lossy_strategy, vault)
+    add_debt_to_strategy(gov, lossy_strategy, vault, amount)
+
+    assert vault.totalAssets() == amount
+    assert vault.strategies(lossy_strategy).current_debt == amount
+
+    amount_to_loose = amount // 2
+    lossy_strategy.setLoss(gov, amount_to_loose, sender=gov)
+    assert asset.balanceOf(lossy_strategy.address) == amount - amount_to_loose
+
+    old_strategy = lossy_strategy
+    new_strategy = create_lossy_strategy(vault)
+
+    old_strategy_params = vault.strategies(old_strategy)
+    old_current_debt = old_strategy_params.current_debt
+    old_max_debt = old_strategy_params.max_debt
+
+    snapshot = chain.pending_timestamp
+    tx = vault.migrate_strategy(new_strategy.address, old_strategy.address, sender=gov)
+
+    event = list(tx.decode_logs(vault.StrategyMigrated))
+
+    assert len(event) == 1
+    assert event[0].old_strategy == old_strategy.address
+    assert event[0].new_strategy == new_strategy.address
+
+    # Funds should still migrate and current debt should stay the same since the loss hasn't been reported
+    new_strategy_params = vault.strategies(new_strategy)
+    assert new_strategy_params.activation == pytest.approx(snapshot, abs=1)
+    assert new_strategy_params.current_debt == old_current_debt
+    assert new_strategy_params.max_debt == old_max_debt
+    assert new_strategy_params.last_report == pytest.approx(snapshot, abs=1)
```

Comment:
I guess it would definitely make sense to use the old activation, since its supposed be just an updated version of the same strategy.

I really like having last_report as a strategy parameter personally. It is useful for off chain monitoring as well. Its a very easy way to do apr calculations or try and find out the next time a strategy is expected to be harvested etc without having to index events or manually tracking. Which could lead to some strategies tracking it and some. Also helpful in strategies for harvestTriggers(tendTriggers in V3?).

Reply:
I guess it should also use the old strategies last_report otherwise any time based fees processed will be incorrect cause it would only charge from migration not the whole time since the last report for strategies that migrate with funds.

Reply:
added 9e21d58cc39ca82304bf9eae3cc3723f8e3ec6d1